src/MCPClient/lib/clientScripts/hasPackages.py

```diff
@@ -19,13 +24,24 @@ def is_extractable(f):
     else:
         return False
 
+def already_extracted(f):
+    """
+    Returns True if this package has already been extracted, False otherwise.
+    """
+    # Look for files in a directory named like <zip file path>-<datetime of extraction>
+    # Don't know exact datetime of extraction, so suffix with '-20' - should be good until year 2100
+    if File.objects.filter(currentlocation__startswith=f.currentlocation + '-20', transfer=f.transfer).exists():
```

Comment:
I'm a little iffy about basing it on the name like this - was it not feasible to base it on the events instead?